### PR TITLE
Update downloads page

### PR DESCRIPTION
**Pull request by ViralBShah**
Remove some redundant stuff that has accumulated over time.

### CODE REVIEW

File: downloads/index.md [hunk 1]
```diff
@@ -263,27 +263,24 @@ Checksums for this release are available in both, [MD5](https://julialang-s3.jul
 ~~~
 Most Linux users should use the glibc binaries unless you know that your system uses musl as its libc.
 
-## Older Releases
+## Older releases
```
Comment (logankilpatrick) [inline downloads/index.md:266]: Is there a specific reason to change these? Won't this break direct links?

**Comment by ViralBShah** [comment]
I've reorganized the GPU stuff in the support table. Ideally we should provide the minimum versions of CUDA, ROCM, and OneAPI in here. 

@vchuravy @maleadt Thoughts on this?

Would we be better off splitting the GPU table from the CPU table, given that it is quite comprehensive by itself now?

File: downloads/index.md (hunk 1 above)
Comment (ViralBShah) [reply downloads/index.md]: I've also deleted some unnecessary section titles. So links are going to break. But this one has been reverted.

File: downloads/index.md [hunk 2]
```diff
@@ -267,23 +267,20 @@ Most Linux users should use the glibc binaries unless you know that your system
 
 Older releases of Julia for all platforms are available on the [Older releases page](/downloads/oldreleases/). Only the LTS and Stable releases are maintained.
 
-## "Nightly" builds
+## Nightly Builds
```
Comment (IanButterworth) [inline downloads/index.md:270]: This was quoted and the text expanded below because people thought they were actually built nightly

Comment (ViralBShah) [reply downloads/index.md:270]: I think we should explain the point in the text, since the quotes are a US-centric thing. It took me a few tries to guess what they were there for.

Comment (ViralBShah) [reply downloads/index.md:270]: Or perhaps we should not call them nightlies, but something like "Binaries sampled from the tip of master"?

Comment (IanButterworth) [reply downloads/index.md:270]: How about
```
## Nightly Builds (not actually nightly)
```
With that I think the existing text below it is ok?

File: downloads/index.md [hunk 3]
```diff
@@ -360,6 +350,21 @@ The platforms currently supported by Julia are listed below. They are divided in
...
+      <td> <a href="https://juliagpu.org/cuda">Nvidia CUDA (64-bit)</a> </td>
+      <td> <font color="green">Tier 1</font> </a> </td>
+    </tr>
+    <tr>
+      <td> ROCM </td>
+      <td> <a href="https://juliagpu.org/rocm">AMD ROCM (64-bit)</a> </td>
+      <td> <font color="crimson">Tier 3</font> </a> </td>
+    </tr>
+    <tr>
+      <td> OneAPI </td>
```
Comment (maleadt) [inline downloads/index.md:364]: The common capitalization is `oneAPI`.

**Comment by maleadt** [comment]
> Ideally we should provide the minimum versions of CUDA, ROCM, and OneAPI in here.

For CUDA that's easy; it requires an NVIDIA driver compatible with CUDA 10.1 or higher. For oneAPI I'm not sure what the exact dependencies are, since we also ship most of the driver (so it'll be a Linux Kernel requirement, depending on the hardware you want to use). And I doubt the story is simple enough for ROCM to fit in a simple table? (cc @jpsamaroo)

File: downloads/index.md (hunk 2 above)
Comment (ViralBShah) [reply downloads/index.md:270]: I added a sentence explaining nightlies.

**Comment by ViralBShah** [comment]
I've added 10.1 for CUDA, and leaving oneAPI and ROCM as is.

**Comment by jpsamaroo** [comment]
AMDGPU ships the right version of ROCm as artifacts, but 4.0.0 is a good version for Julia 1.6, and we'll be shipping 4.2.0 for Julia 1.7 soonish. I'd consider both of those minimum versions that I'm willing to support.

File: downloads/index.md [hunk 4]
```diff
@@ -267,23 +267,20 @@ Most Linux users should use the glibc binaries unless you know that your system
...
+Builds of the current unstable development version of Julia are available on the [nightlies page](/downloads/nightlies/). While the name suggests that they are built every night, they are not strictly nightly. At times, due to broken builds or CI infrastructure issues, nightlies may not be produced. These are intended as developer previews into the latest work and are not intended for normal use. Most users are advised to use the current stable release version of Julia.
 
 ## Download verification
 
 All Julia binary releases are cryptographically secured using the traditional methods on each platform. macOS and Windows releases are codesigned with certificates that are verified by the operating system during installation. Linux and source tarballs are signed with GPG using [this key](/assets/juliareleases.asc).
 
-## Platform specific instructions
+---
 
-Please see [platform specific instructions](/downloads/platform/) for further installation instructions and if you have trouble installing Julia.
```
Comment (fredrikekre) [inline downloads/index.md:280]: Why is this removed? I don't see any other links to these pages

Comment (ViralBShah) [reply downloads/index.md:280]: I thought this is linked in the help parts above.

File: downloads/index.md [hunk 5]
```diff
@@ -267,23 +267,20 @@ Most Linux users should use the glibc binaries unless you know that your system
 
 Older releases of Julia for all platforms are available on the [Older releases page](/downloads/oldreleases/). Only the LTS and Stable releases are maintained.
 
-## "Nightly" builds
+## Nightly Builds
 
-Builds of the current unstable development version of Julia are available on the [nightlies page](/downloads/nightlies/). Built after each update to master, these are intended as developer previews into the latest work and are not intended for normal use. Most users are advised to use the current release version of Julia, above.
+Builds of the current unstable development version of Julia are available on the [nightlies page](/downloads/nightlies/). While the name suggests that they are built every night, they are not strictly nightly. At times, due to broken builds or CI infrastructure issues, nightlies may not be produced. These are intended as developer previews into the latest work and are not intended for normal use. Most users are advised to use the current stable release version of Julia.
```
Comment (IanButterworth) [inline downloads/index.md:272]:
```suggestion
Builds of the current unstable development version of Julia are available on the [nightlies page](/downloads/nightlies/). While the name suggests that they are built every night, they are actually built after each commit to master. However, at times due to broken builds or CI infrastructure issues, nightlies may not be produced. These are intended as developer previews into the latest work and are not intended for normal use. Most users are advised to use the current stable release version of Julia.
```

**Comment by ViralBShah** [comment]
I'm merging this and we can address other concerns in separate issues/PRs.
